modin/pandas/test/dataframe/test_default.py

```diff
@@ -1171,3 +1171,15 @@ def test_hasattr_sparse(is_sparse_data):
         else create_test_dfs(test_data["float_nan_data"])
     )
     eval_general(modin_df, pandas_df, lambda df: hasattr(df, "sparse"))
+
+
+def test_setattr_axes():
+    # Test that setting .index or .columns does not warn
+    df = pd.DataFrame([[1, 2], [3, 4]])
+    with pytest.warns(None):
```

Comment:
I learned while working on something similar today that `pytest.warn(None)` [doesn't check](https://stackoverflow.com/q/45671803/17554722) that we are not issuing warnings. The [recommended way](
https://docs.pytest.org/en/7.0.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests) to check for no warnings is with `warnings.simplefilter`, but we do the equivalent in the entire project's setup.cfg [here](https://github.com/modin-project/modin/blob/38450e10ad13851b61ccc0690c3a98f63b4dc640/setup.cfg#L210), but we skip that check for this entire test [here](https://github.com/modin-project/modin/blob/38450e10ad13851b61ccc0690c3a98f63b4dc640/modin/pandas/test/dataframe/test_default.py#L54) due to #3655.

The end result is that this test doesn't reproduce the bug #4672. Sadly, the [only way](https://stackoverflow.com/a/53363394/17554722) to check that we are NOT warning here but leave the carvout for the rest of the test cases in the file is to do `pytest.mark.parametrize` for every test in the file. We could put every single other test in this file under a new test class, and wrap that class with `@pytest.mark.filterwarnings(default_to_pandas_ignore_string)`.

Let me know if you see a better solution.